cdk/l3-cdk-resources/atlas-basic/src/index.ts

```diff
@@ -1,98 +1,424 @@
+  /**
+     * @description
+     * @type {user.CfnDatabaseUserPropsX509Type}
+     * @memberof DatabaseUserProps
+     */
+  readonly x509Type?: user.CfnDatabaseUserPropsX509Type;
+  /**
+     * @description
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly password?: string;
+  /**
+     * @description
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly projectId?: string;
+  /**
+     * @description
+     * @type {user.RoleDefinition[]}
+     * @memberof DatabaseUserProps
+     */
+  readonly roles?: user.RoleDefinition[];
+  /**
+     * @description
+     * @type {user.ScopeDefinition[]}
+     * @memberof DatabaseUserProps
+     */
+  readonly scopes?: user.ScopeDefinition[];
+  /**
+     * @description
+     * @type {string}
+     * @memberof DatabaseUserProps
+     */
+  readonly username?: string;
+}
+/**
+ * @description
+ * @export
+ * @interface IpAccessListProps
+ */
+export interface IpAccessListProps {
+  /**
+     * @description
+     * @type {ipAccessList.AccessListDefinition[]}
+     * @memberof IpAccessListProps
+     */
+  readonly accessList: ipAccessList.AccessListDefinition[];
+  /**
+     * @description
+     * @type {string}
+     * @memberof IpAccessListProps
+     */
+  readonly projectId?: string;
+  /**
+     * @description
+     * @type {number}
+     * @memberof IpAccessListProps
+     */
+  readonly totalCount?: number;
+  /**
+     * @description
+     * @type {ipAccessList.ListOptions}
+     * @memberof IpAccessListProps
+     */
+  readonly listOptions?: ipAccessList.ListOptions;
+}
+/**
+ * @description
+ * @export
+ * @interface ApiKeyDefinition
+ */
+export interface ApiKeyDefinition {
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly privateKey?: string;
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly publicKey?: string;
+
+}
+/**
+ * @description
+ * @export
+ * @interface AtlasBasicProps
+ * @extends {cdk.StackProps}
+ */
+export interface AtlasBasicProps extends cdk.StackProps{
+  /**
+     * @description
+     * @type {ApiKeyDefinition}
+     * @memberof AtlasBasicProps
+     */
+  readonly apiKeys : ApiKeyDefinition;
+  /**
+     * @description
+     * @type {ProjectProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly projectProps: ProjectProps;
+  /**
+     * @description
+     * @type {ClusterProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly clusterProps: ClusterProps;
+  /**
+     * @description
+     * @type {DatabaseUserProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly dbUserProps?: DatabaseUserProps;
+  /**
+     * @description
+     * @type {IpAccessListProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly ipAccessListProps?: IpAccessListProps;
+}
+/**
+ * @description
+ * @export
+ * @class AtlasBasic
+ * @extends {cdk.Stack}
+ */
 export class AtlasBasic extends cdk.Stack {
```

Comment:
Why are we creating a Stack here and not a Construct? Referring https://github.com/pahud/cdk-mongo-atlas/blob/main/src/atlas-plus.ts from pahud

Reply:
You'll want to vend Constructs, not Stacks. Stacks are meant to be used in apps as logical units of deployment, not in construct libraries.

Reply:
@TheRealAmazonKendra  Thanks for the review, extended Construct.